charts/opentelemetry-demo/values.yaml

```diff
@@ -72,6 +72,10 @@ components:
     podAnnotations: {}
     #  sidecar.opentelemetry.io/inject: "false"
     #  instrumentation.opentelemetry.io/inject-java: "true"
+    schedulingRules:
```

Comment:
We can safely add `schedulingRules` to the redis component as well.  If we make it required in the values.schema.json we can also get rid of the `{{- if .schedulingRules }}` check in `_objects.tpl`.

Reply:
should we make it required? I was not so sure, I would prefer to leave the values.yml in a cleaner state if users dont have specific requirements for allocation

Reply:
By defining it in the default values.yaml we've essentially made it required. Users would have to set it to null in their own values.yaml to get in a space where `.schedulingRules.nodeSelector` blows up on a nil `schedulingRules`.  Explicitly setting it to required avoids that situation.  I don't have strong feelings about it tho.  @dmitryax @Allex1 
 what do you think?

Reply:
I went ahead and made it required, I was under the impression `values.yml` would need to have `schedulingRules` at least as an empty object, but no.